app/models/manager_refresh/inventory_collection.rb

```diff
@@ -368,14 +370,18 @@ class InventoryCollection
     # @param use_ar_object [Boolean] True or False. Whether we need to initialize AR object as part of the saving
     #        it's needed if the model have special setters, serialize of columns, etc. This setting is relevant only
     #        for the batch saver strategy.
+    # @param batch_extra_attributes [Array] Array of symbols marking which extra attributes we want to store into the
+    #        db. These extra attributes might be a product of :use_ar_object assignment and we need to specify them
+    #        manually, if we want to use a batch saving strategy and we have models that populate attributes as a side
+    #        effect.
```

Comment:
for vm, when we change :raw_power_state, other attributes are set, so we need to have

```:batch_extra_attributes => [:power_state, :state_changed_on, :previous_state],```

hopefully this can be autodiscovered later, while passing the same specs